client/client.go

```diff
@@ -60,15 +62,141 @@ type Client struct {
 
 // InvokeResult represents the result of invoking a procedure.
 type InvokeResult struct {
-	Args   wamp.List
-	Kwargs wamp.Dict
-	Err    wamp.URI
+	Args    wamp.List
+	Kwargs  wamp.Dict
+	Err     wamp.URI
+	Options wamp.Dict
 }
 
 // InvocationCanceled is returned from an InvocationHandler to indicate that
 // the invocation was canceled.
 var InvocationCanceled = InvokeResult{Err: wamp.ErrCanceled}
 
+// Helpers for PPT/E2EE
+
+// In mqtt/custom scheme we need to encode payload with specified serializer (if provided)
+func packPPTPayload(options wamp.Dict, args wamp.List, kwargs wamp.Dict) (wamp.List, error) {
+	payload := &wamp.PassthruPayload{
+		Arguments:   args,
+		ArgumentsKw: kwargs,
+	}
+
+	pptSerializerStr, ok := options[wamp.OptPPTSerializer].(string)
+	if ok && pptSerializerStr != "native" {
+
+		var serializer serialize.Serializer
+		pptSerializer, ok := PPTSerializers[pptSerializerStr]
+		if !ok {
+			return nil, ErrPPTSerializerInvalid
+		}
+
+		// Serializer is valid, need to encode payload with it before proceeding
+		switch pptSerializer {
+		case JSON:
+			serializer = &serialize.JSONSerializer{}
+		case MSGPACK:
+			serializer = &serialize.MessagePackSerializer{}
+		case CBOR:
+			serializer = &serialize.CBORSerializer{}
+			// In future should be extended with FlatBuffers
+		}
+
+		bin, err := serializer.SerializeDataItem(payload)
+		if err != nil {
+			return nil, ErrSerialization
+		}
+
+		return wamp.List{bin}, nil
+	}
+
+	return wamp.List{payload}, nil
+}
+
+func unpackPPTPayload(details wamp.Dict, args wamp.List) (wamp.List, wamp.Dict, error) {
+	payloadTyped := &wamp.PassthruPayload{}
```

Comment:
```suggestion
	var pptPayload wamp.PassthruPayload
```

or to look similar to code around that uses `:=` 
```
pptPayload := wamp.PassthruPayload{}
```